extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -281,9 +281,7 @@ private String normalizeLocation(String location) {
     }
 
     private FileSystem initFileSystem(final URI uri) throws IOException {
-        final Map<String, String> env = new HashMap<>();
-        env.put("create", "true");
-        return FileSystems.newFileSystem(uri, env);
+        return ZipUtils.newZip(Path.of(uri));
```

Comment:
@Postremus this change is problematic. The hibernate-orm-multi-tenancy-quickstart is now failing with:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project hibernate-orm-multi-tenancy-quickstart: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.flyway.FlywayProcessor#build threw an exception: java.nio.file.FileSystemNotFoundException
[ERROR] 	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:169)
[ERROR] 	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:155)
[ERROR] 	at java.base/java.nio.file.Path.of(Path.java:208)
[ERROR] 	at io.quarkus.flyway.FlywayProcessor.initFileSystem(FlywayProcessor.java:285)
[ERROR] 	at io.quarkus.flyway.FlywayProcessor.discoverApplicationMigrations(FlywayProcessor.java:231)
[ERROR] 	at io.quarkus.flyway.FlywayProcessor.build(FlywayProcessor.java:99)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:887)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Could you have a look very soon? We need a fix before next Tuesday.

Thanks!

Reply:
And same for the `quartz-quickstart`.

Reply:
I'm wondering whether it's related to URI -> Path -> URI translation.

Reply:
Failing path is `jar:file:/data/home/gsmet/git/quarkus-quickstarts/quartz-quickstart/target/quartz-quickstart-1.0.0-SNAPSHOT.jar!/db/migration/` for Quartz.

I have no idea if it's correct and should contain the `!/db/migration/` part. Also, I don't know if the `env.put("create", "true");` if of interest.

Reply:
I created https://github.com/quarkusio/quarkus/issues/22079